### PR TITLE
test: kubectl-dns secret-generation command

### PR DESCRIPTION
**Pull request by averevki**
## Description
  - Add support for `kubectl-dns` CLI tool integration to the test suite
  - Implement wrapper class and test for the `kubectl-dns secret-generation` command
  - Refactor CLI tooling by moving `kuadrantctl` into a dedicated `testsuite/cli/` module

  ## Verification steps

  Run the new test:
  ```bash
make testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py
 ```
  Note: This test requires:
  - The [kubectl-dns binary](https://github.com/Kuadrant/dns-operator/releases) to be available in PATH or configured in config/settings.local.yaml
  - A multicluster CoreDNS setup with one primary and one secondary cluster
  -  2 kubeconfigs containing `certificate-authority` key with the cluster CA. You can set it for a kubeconfig by logging in with explicitly specified path to the cluster CA, aka. `oc login --username=${} --password=${} <clusterapiurl> --certificate-authority=path/to/ca`

## Release Notes

* **New Features**
  * Added support for multi-cluster CoreDNS secret generation using kubectl-dns.

* **Tests**
  * Introduced test infrastructure and markers for command-line tool validation.
  * Added comprehensive test suite for multi-cluster DNS secret generation scenarios, including kubeconfig merging across clusters.

* **Chores**
  * Updated default configuration to include kubectl-dns settings.

### CODE REVIEW

File: testsuite/kubernetes/client.py [hunk 1]
```diff
@@ -142,3 +151,25 @@ def apply_from_string(self, string, cls, cmd_args=None):
             obj = selector.object(cls=cls)
             obj.context = self.context
         return obj
+
+    def create_merged_kubeconfig(self, cluster2: "KubernetesClient") -> str:
```
Comment (azgabur) [inline testsuite/kubernetes/client.py]: Due to coredns tests we will probably deal with more then two clusters, maybe refactor this method to take list of additional clusters? wdyt?

File: testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    if not shutil.which(binary_path):
+        skip_or_fail("kubectl-dns binary not found")
+    return KubectlDNS(binary_path)
+
+
+@pytest.fixture(scope="module")
+def kubeconfig_secrets(request, testconfig, cluster, cluster2, kubectl_dns, blame):
+    """Run generate-secret command on merged kubeconfig to generate kubeconfig secret for the secondary cluster"""
+    system_project = testconfig["service_protection"]["system_project"]
+    secret_name = blame("kubecfg")
```
Comment (azgabur) [inline testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py]: Would use `tempfile` library for a temporary file.  Also there is missing finalizer for removing it.

**Review by azgabur — COMMENTED** [review]
Did not run the test yet

File: testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py (hunk 2 above)
Comment (averevki) [reply testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py]: Binary creates this secret directly on cluster. I have finalizer set right under this comment

File: testsuite/kubernetes/client.py [hunk 3]
```diff
@@ -147,3 +156,25 @@ def apply_from_string(self, string, cls, cmd_args=None):
...
+        with cluster2.context:
+            config2_yaml = oc.invoke("config", ["view", "--minify=true", "--flatten=true"]).out()
+
+        config1 = yaml.safe_load(config1_yaml)
+        config2 = yaml.safe_load(config2_yaml)
+
+        merged_config = config1.copy()
+        merged_config["clusters"].extend(config2.get("clusters", []))
+        merged_config["contexts"].extend(config2.get("contexts", []))
+        merged_config["users"].extend(config2.get("users", []))
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/kubernetes/client.py:176]: `user` is the same, but not sure this is a problem :thinking: Would look how the merged config looks like though

File: testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import shutil
+
+import dns.resolver
+import pytest
+
+from testsuite.cli.kubectl_dns import KubectlDNS
+from testsuite.tests.multicluster.coredns.conftest import IP1, IP2
+
+pytestmark = [pytest.mark.cli]
```
[the remark this answers is not in the document]
Comment (azgabur) [reply testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py:11]: cli testing is fine that is separate

File: testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+def kubectl_dns(testconfig, skip_or_fail):
+    """Return Kuadrantctl wrapper with merged kubeconfig"""
+    binary_path = testconfig["kubectl-dns"]
+    if not shutil.which(binary_path):
+        skip_or_fail("kubectl-dns binary not found")
+    return KubectlDNS(binary_path)
+
+
+@pytest.fixture(scope="module")
+def kubeconfig_secrets(request, testconfig, cluster, cluster2, kubectl_dns, blame):
```
[the remark this answers is not in the document]
Comment (averevki) [reply testsuite/tests/multicluster/coredns/two_clusters/kubectl_dns/test_secret_generation.py:24]: `kubeconfig_secrets` fixture is requested and commited from commit fixture in `coredns/two_clusters/` conftest, which has autouse on it

File: testsuite/kubernetes/client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (averevki) [reply testsuite/kubernetes/client.py:176]: There are multiple places I expect testsuite to fail if this happens, finally it would fail on dns cli command execution if context name doesn't exist or the same cluster is used for both kubeconfigs. Cluster names are uniquely generated if you log in with `oc` or `ocp`, and context and user names are constructed from the cluster name. Right now there are a few prerequisites in order to execute this test, including a kubeconfig with CA authority referenced in it. Test will be refactored in the future, right now I think it is an edge case not worth to care about. And even if there is something wrong with kubeconfigs, we would still know from this test in the current state
